### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by yosefHesham**
### Brief of what changed:

Added the functionality to call API with get and post methods

---

**Changes Made:**

- Created leaderboard class service to handle API requests.
- Created getData method.
- Created sendData method
- Created module to handle HTML dynamically.

### CODE REVIEW

File: modules/leaderboard_ui.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  for (let i = 0; i < data.length; i += 1) {
+    const element = createElement(data[i], i % 2 === 0);
+    setTimeout(() => {
+      list.appendChild(element);
+    }, 400 * i);
+  }
+};
+
+refreshButton.addEventListener('click', () => {
```
Comment (FrederickMih) [inline modules/leaderboard_ui.js:32]: - [ ] When I click the refresh button after submitting my entries the score list doesn't alter. No user data is reflected on the scoreboard.  Please rectify this issue.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @yosefHesham , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



# To Highlights 

:fast_forward: GitHub workflow has been followed :heavy_check_mark: 
:fast_forward: PR has a title and summary :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/leaderboard_ui.js (hunk 1 above)
Comment (yosefHesham) [reply modules/leaderboard_ui.js:32]: are u sure? 
take another look please 
@FrederickMih

Comment (FrederickMih) [reply modules/leaderboard_ui.js:32]: > are u sure? take another look please @FrederickMih

Hi @yosefHesham, can you connect with me through [slack]([@Fred Mih](https://microverse-crs.slack.com/team/U02N2KH5KD4)) let's take a look together?.

Comment (yosefHesham) [reply modules/leaderboard_ui.js:32]: @FrederickMih  okay i will send u after 30 min, but anyways, i deployed the page and it works in the deployed page.. check again please:
https://yosefhesham.github.io/leaderboard/dist/

Comment (FrederickMih) [reply modules/leaderboard_ui.js:32]: Hi @yosefHesham , you can request a review once more.

Comment (yosefHesham) [reply modules/leaderboard_ui.js:32]: @FrederickMih  that means it works?

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @yosefHesham , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/leaderboard_ui.js (hunk 1 above)
Comment (yosefHesham) [reply modules/leaderboard_ui.js:32]: @FrederickMih thanks for wasting my time..

**Pull request merged**